EightMinuteEmpireLegends/MapLoader/MapLoader.cpp

```diff
@@ -0,0 +1,136 @@
+#pragma once
+#include "MapLoader.h"
+#include "Map.h"
+#include <fstream>
+#include <iostream>
+#include "json.hpp"
+
+using std::string;
+using std::vector;
+using json = nlohmann::json;
+using std::cout;
+
+MapLoader::MapLoader() {
+
+}
+
+bool MapLoader::validateMap(std::string path) {
+	std::ifstream fileContentStream(path);
+	bool isValid = json::accept(fileContentStream);
+
+	if (!isValid) {
+		std::cout << "The file is not a valid json file.\n";
+	}
+
+	return isValid;
+}
+
+Map* MapLoader::parseMap(string path) {
+	Map* mapPtr = NULL;
+
+	//Make sure that we have a json file
+	if (validateMap(path)) {
+		std::ifstream fileContentStream(path);
+		json mapJson = json::parse(fileContentStream);
+
+		try {
+			//Create the list of continents first
+			json jsonContinents = mapJson["continents"];
+			int numOfContinents = static_cast<int>(jsonContinents.size());
+
+			vector<Continent*> continents;
+
+			for (int i = 0; i < numOfContinents; i++) {
+				auto jsonContinent = jsonContinents[i];
+				string continentName = jsonContinent["name"].get<std::string>();
+
+				Continent* continentPtr = new Continent();
+				continentPtr->name = continentName;
+
+				continents.push_back(continentPtr);
+			}
+
+			//Create the map with the continents
+			string mapName = mapJson["name"];
+			mapPtr = new Map(mapName, continents);
+
+			//Create the list of vertices and their correspoding territories
+			json jsonVertices = mapJson["vertices"];
+			int numOfVertices = static_cast<int>(jsonVertices.size());
+
+			for (int j = 0; j < numOfVertices; j++) {
+				auto jsonVertex = jsonVertices[j];
+				string id = jsonVertex["id"].get<std::string>();
+				string territoryName = jsonVertex["territoryName"].get<std::string>();
+				string continentName = jsonVertex["continentName"].get<std::string>();
+
+				Continent* continentPtr = getContinentByName(continents, continentName);
+				if (continentPtr == NULL)
+					throw "The continent with the name <<" + continentName + ">> does not exist";
```

Comment:
im not too sure here, but maybe its not a good practice:
https://stackoverflow.com/questions/134569/c-exception-throwing-stdstring

Reply:
Yeah I am not sure either, but I just want to be sure that the error is handled since I didn't have much time to research on this.  If we have time I can improve the code.